lab_3_generate_by_ngrams/main.py

```diff
@@ -388,6 +609,34 @@ def run(self, prompt: str, seq_len: int) -> Optional[str]:
         In case of corrupt input arguments or methods used return None,
         None is returned
         """
+        if not isinstance(seq_len, int) or seq_len <= 0 or not isinstance(prompt, str) or not prompt:
+            return None
+
+        encoded_prompt = self._text_processor.encode(prompt)
+        if encoded_prompt is None:
+            return None
+        candidates = {encoded_prompt: 0.0}
+
+        for _ in range(seq_len):
+            updated_candidates = candidates.copy()
+
+            for sequence in candidates:
+                next_tokens = self._get_next_token(sequence)
+                if not next_tokens:
+                    return None
+
+                possible_sequences = self.beam_searcher.continue_sequence(sequence, next_tokens, updated_candidates)
+                if not possible_sequences:
+                    return self._text_processor.decode(sorted(tuple(
```

Comment:
consider breaking the cycle instead of duplicating code

Reply:
fixed

Reply:
I've changed this part of code a bit so tests won't pass with breaking cycle here(